src/clusters.ts

```diff
@@ -72,7 +72,7 @@ export async function getClusterDetails(vendorPortalApi: VendorPortalApi, cluste
       }
     
       const body: any = JSON.parse(await res.readBody());
-      if (clusters.length == body.totalClusters || body.clusters.length == 0) {
+      if (clusters.length == body.totalClusters || body.clusters == undefined) {
```

Comment:
would something like `body.clusters?.length == 0` handle both cases where it's undefined and empty?